src/providers/auth.service.ts

```diff
@@ -87,15 +87,15 @@ export class AuthService {
    * @param unlock
    * @returns {Observable|"../../Observable".Observable|"../../../Observable".Observable}
    */
-  public otp(data, unlock){
+  public otp(data, unlock) {
     let response = new Observable((observer) => {
-      this.authHttp.post(environment.server_url + "/api/auth/otp" + ((unlock)? "/unlock" : ""), data)
+      this.authHttp.post(environment.server_url + "/api/auth/otp" + ((unlock) ? "/unlock" : ""), data)
```

Comment:
Why create a variable for response? Why not just return the Observable immediately?